web-services-app/src/main/java/edu/unc/lib/boxc/web/services/processing/DownloadImageService.java

```diff
@@ -128,14 +141,51 @@ private String getExtent(ContentObjectRecord contentObjectRecord) {
         return ds.getExtent();
     }
 
-    private int getLongestSide(ContentObjectRecord contentObjectRecord) {
+    /**
+     * If the shortest side of the content object is smaller than the size requested, return the placeholder
+     * @param contentObjectRecord
+     * @param size
+     * @return true if service needs to return a placeholder image
+     */
+    private boolean needsPlaceholder(ContentObjectRecord contentObjectRecord, String size) {
+        if (Objects.equals(FULL_SIZE, size)) {
+            return false;
+        }
+        var shortestSide = getShortestSide(contentObjectRecord);
+        return shortestSide < Integer.parseInt(size);
+    }
+
+    private String[] getImageDimensions(ContentObjectRecord contentObjectRecord) {
```

Comment:
It would make sense for this to return `int[]`, and would centralize the parseInt calls.